2003/kangraemin/app/src/main/java/com/example/kangraemin/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,133 @@
+package com.example.kangraemin.ui.login
+
+import android.content.Intent
+import android.os.Bundle
+import com.example.kangraemin.ui.main.MainActivity
+import com.example.kangraemin.R
+import com.example.kangraemin.base.KangBaseActivity
+import com.example.kangraemin.model.AppDatabase
+import com.example.kangraemin.model.AuthRepository
+import com.example.kangraemin.model.local.datadao.AuthLocalDataSourceImpl
+import com.jakewharton.rxbinding3.view.clicks
+import com.jakewharton.rxbinding3.view.focusChanges
+import com.jakewharton.rxbinding3.widget.textChanges
+import io.reactivex.Observable
+import io.reactivex.functions.BiFunction
+import kotlinx.android.synthetic.main.activity_login.*
+
+class LoginActivity : KangBaseActivity(), LoginContract.View {
+
+    private lateinit var presenter: LoginContract.Presenter
+
+    private val authRepository by lazy {
+        val db = AppDatabase.getInstance(context = this)
+        AuthRepository(authLocalDataSource = AuthLocalDataSourceImpl(db.authDao()))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        presenter = LoginPresenter(this)
+
+        presenter.checkAutoLoginStatus(authRepository = authRepository)
+
+        val checkLoginInfoHasEntered = Observable
+            .combineLatest(
+                et_id.textChanges(),
+                et_pw.textChanges(),
+                BiFunction { id: CharSequence, password: CharSequence ->
+                    presenter
+                        .checkLoginInfoHasEntered(
+                            id = id.toString(),
+                            password = password.toString()
+                        )
+                }
+            )
```

Comment:
이 로직은 굳이 번거롭게 연산자를 쓰기 보다 
```suggestion
   presenter.checkLoginInfoHasEntered(
     id = et_id.text.toString(),
     password = et_pw.text.toString()
   )
```
이렇게 가는게 더 보기 좋을 것 같아요.

Reply:
이부분 같은 경우는 여쭤볼게 있어 코멘트 남깁니다 

전 텍스트가 바뀔 때 마다 모든 값들이 입력되었는지 확인하고자 하였습니다. 
그런데 compositLatest로 textChange 스트림을 받지 않으면 텍스트가 바뀔 때 마다 체크를 못하게 되는것 같아 어떻게 고쳐야 할 지 감이 잘 안잡혀 여쭤봅니다 

아래의 부분도 해당 부분을 수정 할 때 같이 고치도록 하겠습니다.

Reply:
@kangraemin 음.. 그렇다면 View에서는 단어가 변경될 때마다 presenter에 넘겨주는 역할만 하고,
나머지는 그 후 presenter에서 판단하에 View에 처리를 요청하도록 해야될 것 같아요.

Reply:
수정하였습니다
7e30d7f